storage_controller/src/service.rs

```diff
@@ -7919,6 +7922,15 @@ impl Service {
             "Auto-splitting tenant for size threshold {split_threshold}: current size {split_candidate:?}"
         );
 
+        // Retain the stripe size of sharded tenants, as splits don't allow changing it. Otherwise,
+        // use DEFAULT_STRIPE_SIZE for unsharded tenants -- their stripe size doesn't really matter,
+        // and if we change the default stripe size we want to use the new default rather than an
+        // old, persisted stripe size.
+        let new_stripe_size = match split_candidate.id.shard_count.count() {
+            0 | 1 => Some(ShardParameters::DEFAULT_STRIPE_SIZE),
```

Comment:
nit: `count()` should never return 0 -- would rather have an assertion branch here so that we don't give the impression that count()==0 is A Thing.

(This is a piece of tech debt that the type returned is not a NonZero type)

Reply:
Sure, done.